src/sagemaker/processing.py

```diff
@@ -1195,7 +1195,9 @@ def __init__(
             source (str): The source for the output.
             destination (str): The destination of the output. If a destination
                 is not provided, one will be generated:
-                "s3://<default-bucket-name>/<job-name>/output/<output-name>".
+                "s3://<default-bucket-name>/<job-name>/output/<output-name>"
+                (Note: this does not apply when used with
+                :class:`~sagemaker.workflow.steps.ProcessingStep`).
```

Comment:
what will be used if its pipeline session, or processing step?

Reply:
It should be `s3://<default-bucket-name>/<job-name created by the processor class>/output/<output-name>`

The `job-name created by the processor class` here will be popped out from the DSL and is not the actual job name (created in Pipeline backend). To avoid confusing users, I don't give this path here.